notmuch/db.c

```diff
@@ -102,6 +103,18 @@ notmuch_database_t *nm_db_do_open(const char *filename, bool writable, bool verb
   const notmuch_database_mode_t mode =
       writable ? NOTMUCH_DATABASE_MODE_READ_WRITE : NOTMUCH_DATABASE_MODE_READ_ONLY;
 
+  struct stat st1;
+  struct stat st2;
+  char buf[PATH_MAX];
+  if (stat(filename, &st1) != 0)
+    mutt_error(_("Can't stat %s: %s"), filename, strerror(errno));
+  else if (!S_ISDIR(st1.st_mode))
+    mutt_error(_("%s is not a directory"), filename);
+  if (stat(mutt_path_concat(buf, filename, ".notmuch", sizeof(buf)), &st2) != 0)
+    mutt_error(_("Can't stat %s: %s"), filename, strerror(errno));
+  else if(!S_ISDIR(st2.st_mode))
+    mutt_error(_("%s is not a directory"), buf);
+
```

Comment:
Just a general comment, but we should reword the actual error messages to let the user know it's due to notmuch. Maybe something along the lines of `Unable to open notmuch database: can't stat /some/path`?

Reply:
@jindraj reused existing messages to avoid adding extra translations (there are 1500 already! :-)

Reply:
Hmm, that's a good point regarding translations. 

However, if I have `mailboxes "notmuch:///?query=tag:inbox` (note the extra `/`), it produces a rather unhelpful message `Can't stat /: No such file or directory`.  That could be caused by anything.

Perhaps it would be best to always stat the `.notmuch` folder so the message is `Can't stat /.notmuch: No such file or directory`. That at least gives the user an indication it's notmuch-related without increasing translation count.